File: php/HighSchoolSweetheart.php
<?php

class HighSchoolSweetheart
{
    public function firstLetter(string $name): string
    {
        return ltrim($name)[0];
    }

    public function initial(string $name): string
    {
        $initial = strtoupper($this->firstLetter($name));

        return "{$initial}.";
    }

    public function initials(string $name): string
    {
        $nameArr = explode(" ", trim($name));
        $initials = '';

        foreach ($nameArr as $nameSplit) {
            $initials .= $this->initial($nameSplit) . ' ';
        }

        return rtrim($initials);
    }

    public function pair(string $sweetheart_a, string $sweetheart_b): string
    {
        $firstInitials = $this->initials($sweetheart_a);
        $secondInitials = $this->initials($sweetheart_b);

        return <<<HEART
                 ******       ******
               **      **   **      **
             **         ** **         **
            **            *            **
            **                         **
            **     $firstInitials  +  $secondInitials     **
             **                       **
               **                   **
                 **               **
                   **           **
                     **       **
                       **   **
                         ***
                          *
            HEART;
    }
}

File: elixir/high_school_sweetheart.ex
defmodule HighSchoolSweetheart do
  def first_letter(name) do
    name
    |> String.trim()
    |> String.first()
  end

  def initial(name) do
    initial =    
      name
      |> first_letter()
      |> String.capitalize()

    "#{initial}."
  end

  def initials(full_name) do
    initials =
      full_name
      |> String.split()
      |> Enum.map(&initial/1)

    "#{hd(initials)}" <> " " <> "#{tl(initials)}"
  end

  def pair(i1, i2) do
"""
     ******       ******
   **      **   **      **
 **         ** **         **
**            *            **
**                         **
**     #{initials(i1)}  +  #{initials(i2)}     **
 **                       **
   **                   **
     **               **
       **           **
         **       **
           **   **
             ***
              *
"""
  end
end
